lib/volk_malloc.c

```diff
@@ -29,102 +29,30 @@
 /*
  * For #defines used to determine support for allocation functions,
  * see: http://linux.die.net/man/3/aligned_alloc
+ *
+ * C11 features:
+ * see: https://en.cppreference.com/w/c/memory/aligned_alloc
 */
 
-// Otherwise, test if we are a POSIX or X/Open system
-// This only has a restriction that alignment be a power of 2 and a
-// multiple of sizeof(void *).
-#if _POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600 || HAVE_POSIX_MEMALIGN
 
 void *volk_malloc(size_t size, size_t alignment)
 {
-  void *ptr;
-
-  // quoting posix_memalign() man page:
-  // "alignment must be a power of two and a multiple of sizeof(void *)"
-  // volk_get_alignment() could return 1 for some machines (e.g. generic_orc)
-  if (alignment == 1)
+  if (alignment == 1){
```

Comment:
(1) Thinking to just remove this statement, since one can use `aligned_alloc` with alignment of 1. I'd guess internal to that function it checks the alignment & if == 1 then uses `malloc`, so I'd bet we're not saving a lot of CPU cycles here. And, it cleans up our code.
(2) Thinking making both `volk_malloc` and `volk_free` inline functions, since they are so short to begin with; no real need to have them be proper functions, IMHO. Maybe `__VOLK_ATTR_INLINE` since that already exists in `volk_common.h`? If not that, then we can define a `__VOLK_FUNC_INLINE` or something like it & use that here.

Reply:
Since GCC throws this warning, I'd suggest we don't inline this function. 
```
warning: always_inline function might not be inlinable [-Wattributes]
 __VOLK_ATTR_INLINE void volk_free(void *ptr)
                         ^~~~~~~~~
warning: always_inline function might not be inlinable [-Wattributes]
 __VOLK_ATTR_INLINE void *volk_malloc(size_t size, size_t alignment)
```

That would also be consistent with its current implementation. Further, I'd assume current compilers inline such functions whenever appropriate and possible.